api4/post_test.go

```diff
@@ -1216,15 +1216,37 @@ func TestGetPostsForChannel(t *testing.T) {
 		})
 
 		// the endpoint should work fine when viewing archived channels is enabled
-		_, _, err = c.GetPostsForChannel(channel.Id, 0, 10, "", false)
+		_, _, err = c.GetPostsForChannel(channel.Id, 0, 10, "", false, false)
 		require.NoError(t, err)
 
 		// the endpoint should return forbidden if viewing archived channels is disabled
 		th.App.UpdateConfig(func(cfg *model.Config) { *cfg.TeamSettings.ExperimentalViewArchivedChannels = false })
-		_, resp, err = c.GetPostsForChannel(channel.Id, 0, 10, "", false)
+		_, resp, err = c.GetPostsForChannel(channel.Id, 0, 10, "", false, false)
 		require.Error(t, err)
 		CheckForbiddenStatus(t, resp)
 	}, "Should forbid to retrieve posts if the channel is archived and users are not allowed to view archived messages")
+
+	client.DeletePost(post10.Id)
+	client.DeletePost(post8.Id)
+
+	// include deleted posts for non-admin users.
+	_, resp, err = client.GetPostsForChannel(th.BasicChannel.Id, 0, 100, "", false, true)
+	require.Error(t, err)
+	CheckForbiddenStatus(t, resp)
+
+	th.TestForSystemAdminAndLocal(t, func(t *testing.T, c *model.Client4) {
+		// include deleted posts for admin users.
```

Comment:
Local mode is an unrestricted mode that works through a local socket connection. Basically any one that has access to the local file system (and thus the socket connection), can call API methods with admin privileges. 

https://docs.mattermost.com/manage/mmctl-command-line-tool.html#local-mode

Let's not change this test, because we also want to confirm the changes work in local mode.